### ISSUE TITLE
Lua table support

### ISSUE THREAD

**Issue by Zash** [mikefarah/yq#1700]
**Please describe your feature request.**
A clear and concise description of what the request is and what it would solve. 
Eg. I wish I could use yq to support [Lua](https://www.lua.org/) [tables](https://www.lua.org/manual/5.4/manual.html#3.4.9) as input and output formats. This would allow querying and manipulating Lua based file formats like [LuaRocks](https://luarocks.org/) [rockspec format](https://github.com/luarocks/luarocks/wiki/Rockspec-format) or the [Prosody](https://prosody.im/) `internal` data storage format.

**Describe the solution you'd like**
If we have data1.yml like:
(please keep to around 10 lines )

```yaml
country: Australia # this place
cities:
- Sydney
- Melbourne
- Brisbane
- Perth
```

And we run a command:

```bash
yq --output-format lua data1.yml
```

it could output

```lua
country = "Australia" -- this place
cities = {
  "Sydney",
  "Melbourne",
  "Brisbane",
  "Perth",
}
```

or alternatively (perhaps a flag)

```lua
return {
  country = "Australia"; -- this place
  cities = {
    "Sydney",
    "Melbourne",
    "Brisbane",
    "Perth",
  }
}
```

Lua tables support arbitrary types as both keys and values, but in stored serialized formats this is rarely used, so keeping to a subset such as only allowing scalar keys may be sensible.

**Describe alternatives you've considered**
Writing simple converters to/from Lua and yaml/json etc is relatively easy, I have a few of these already, but they lack the query and manipulation capabilities and usually don't preserve comments or order of keys like `yq` does.

**Comment by mikefarah** [comment]
Fixed in 4.35.1

**Comment by Zash** [comment]
Input support would be cool too.
